lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -118,7 +117,11 @@ getUsdAttributeValueAsString(const PXR_NS::UsdAttribute& attr, const PXR_NS::Usd
         return os.str();
     }
 
-    UFE_ASSERT_MSG(false, kErrorMsgFailedConvertToString);
+    try {
+        UFE_ASSERT_MSG(false, kErrorMsgFailedConvertToString);
+    } catch (const std::exception&) {
+        // noop
+    }
```

Comment:
As we discussed we should be trapping all these asserts since they will throw. I also removed any extraneous ones about get first before set to verify the type. We'll let USD handle that.